proposals/extend_pod_security_admission.md

```diff
@@ -0,0 +1,236 @@
+           # +required
+         - controlName: Capabilities
+           images:
+           - ghcr.io/example/nginx:1.2.3
+```
+
+If the user wants to specifically skip checking the custom "add" capabilities, `restrictedField` can be used to define the allowed list. For example, the following rule applies Restricted PSS to namespace "test" and "staging" while excluding pods running image "ghcr.io/example/nginx:1.2.3" to be able to add `kill`, `SETGID` and `SETUID` capabilities. 
+
+```yaml
+validationFailureAction: enforce
+rules:
+  - name: validate-host-network
+    match:
+      any:
+      - resources:
+          kinds:
+          - Pod
+        namespaces:
+        - test
+        - staging
+    exclude:
+      any:
+      - userInfo:
+          username: dummyuser
+    validate:
+      # this new type of rule only deals with PSS profiles
+      # as we need to check if the value in the resource
+      # is actually allowed by PSS
+      podSecurity:
+         profile: restricted
+         # version must be a valid Kubernetes minor version,
+         # or `latest`.
+         version: latest
+         exclude:
+           # controlName is the Control defined in PSS
+           # +required
+         - controlName: Capabilities
+           images:
+           - ghcr.io/example/nginx:1.2.3
+           # +optional
+           restrictedField: spec.containers[*].securityContext.capabilities.add
+           # Any type
+           # +optional
+           values: # how to define an object?
+           - "KILL"
+           - "SETGID"
+           - "SETUID"
+```
+
+Note that securityContext is configured at container level, not pod, as the matching image selects a specific container. The pod creation will be rejected if there's any violation found in its container. For pod level checks, other match criteria such as labels could be used.
+
+For a PolicyReport, `cpol.rules.validate.podSecurity.exclude.restrictedField` and `cpol.rules.validate.podSecurity.exclude.controlName` will be added to `polr.results.properties` to audit the result.
+
+```yaml
+PolicyReport:
+results:
+- message: |- 
+    'validation error: The label `app.kubernetes.io/name` is required. 
+    Rule check-for-labels failed at path
+    /metadata/labels/app.kubernetes.io/name/'
+  policy: require-labels
+  resources:
+  - apiVersion: v1
+    kind: Pod
+    name: nginx
+    namespace: default
+    uid: 13fcb726-5597-4fcd-a708-ce8b558bd484
+  rule: check-for-labels
+  properties:
+    controlName: Capabilities
+    podSecurity: spec.containers[*].securityContext.capabilities.add
+  scored: true
+  status: fail
+```
+
+### Exceptions for container security context
+
+* Host Namespaces: `spec.hostNetwork,spec.hostPID,spec.hostIPC`
+* Sysctls: `spec.securityContext.sysctls[*].name`
+* volumes: `spec.volumes`
+* Non-root groups: `spec.securityContext.supplementalGroups[*], spec.securityContext.fsGroup`
+
+
+## Alternate Solutions Considered
```

Comment:
To echo [Marcel's comment above](https://github.com/kyverno/KDP/pull/22/files#r901378337) and [my own Slack thread](https://kubernetes.slack.com/archives/CLGR9BJU9/p1645606236122979), this PSA extension doesn't allow much room for enabling self-service or internal platforms, especially under Gitops workflows.

With PSPs, any user/SA can be authorized to bring their own PSPs, bound by RBAC.
With Kyverno PSS policies/PSA extension, all exceptions must be stored in the policy itself, so either many users will need update access to ClusterPolicies or the ability to modify and deploy policies from a backing repository. Those are clearly not ideal for security, so most likely a small group of people will be required to be involved in all PSS exceptions. In many organizations/environments this is fine and perhaps ideal, but for others it will not scale, and I don't see a nice way for cluster admins to somehow abstract and extend exception functionality to other teams.

So my questions then are:
Is this section, particularly `Extend Via Annotation` to suggest that some annotation-based workflow is planned to be supported in the future, or rather that it was considered and rejected?

What would be the official suggestion for allowing some degree of self-service with this model?

Reply:
Relevant to this comment is an [existing feature request](https://github.com/kyverno/kyverno/issues/2627) for predefinition of policy violation exceptions.

Reply:
`Extend Via Annotation` was considered and rejected. Will stick with the proposed solution `Extend PSA via a new rule`.